feat: Include stage and job in alert name in GoCD alerting

timmc-edx opened the pull request:
These alerts are specific to a job, so the name should reflect this. (Otherwise we get multiple alerts for different parts of a pipeline and can't tell which is which.)

macdiesel commented:
@doctoryes why not just take restrict_to_stage and validate to the ec2 module and just have the scripts call them from the ec2 module? I think it would make the code a bit cleaner and you are already testing it in the test_ec2 module anyway.

doctoryes commented:
@macdiesel Given the limitations in being able to test the scripts, your proposal makes sense. This restriction will end up pushing all script functionality into core python code, which I guess isn't a bad thing - the scripting layer could fall away without losing most of the functionality it implements. Bottom line: I'll do so.

doctoryes commented:
@macdiesel Done. Ready for re-review.

macdiesel commented:
👍 🚢 🇮🇹

timmc-edx committed:
feat: Include stage and job in alert name in GoCD alerting

These alerts are specific to a job, so the name should reflect this.
(Otherwise we get multiple alerts for different parts of a pipeline and
can't tell which is which.)
